### ISSUE TITLE
Handling of extraneous data in test suite using strictness parameter

### ISSUE THREAD

**Issue by kdahlquist** [dondi/GRNsight#339]
Working on implementing a strictness parameter for the handling of extraneous data in the network sheet according to the discussion from 9/26.

**Comment by kdahlquist** [comment]
Since the next task for @mihirsamdarshi and @yshin4 will be to refactor the test suite for graphml and sif, #300, they should work with @NAnguiano and @anuvarsh on #339 so that they can get familiar with the test suite.

**Comment by kdahlquist** [comment]
The current workaround is that @anuvarsh and @NAnguiano made a cap on the number of errors that the server could handle.  Right now it's set at 1000, but it could be changed.  Actually, I think anything over like 20 errors is sufficient to stop.  No one is going to look at so many errors.

We need to put language into the error message that says:

Your file generated >= x errors, so GRNsight stopped checking.  If you fix these errors and try to upload again, there may be further errors detected.

**Comment by NAnguiano** [comment]
The cap on the number of errors somehow disappeared during the merge, so I reimplemented it with a cap of 20 errors. There isn't currently any text indicating that there is too many errors. 

It's worth noting that this invalidates the warning message of the extraneousDataWarning. Since a large number of emptyRowErrors typically indicate extraneous data, the errors are removed and replaced with the extraneous data warning, which takes the row of the final emptyRowError and uses that to indicate the most likely row the extraneous data is in. This will likely eventually need to be changed.

**Comment by anuvarsh** [comment]
@NAnguiano updated the strictness to 20 errors, and I added the updated error message. 

Because emptyRowErrors is the only error that doesn't immediately return the sheet (i.e. it continues checking the sheet for additional errors) the strictness parameter and error message were applied only to this error. Though this solved the problem temporarily, if in the future we added another error that did not return the network immediately, the error cap would not apply.

In order to prevent problems for future-GRNsight, I am writing a generalized errors count function to mimic the warnings count function which limits the errorsList from ever being more than 20 errors long. 

Current status: 4 tests are failing because of a "Maximum call stack size is exceeded" error, most likely because of a circular function. Hopefully will fix quickly.
